ProjectLighthouse.Servers.GameServer/Controllers/Slots/ScoreController.cs

```diff
@@ -116,12 +118,19 @@ public async Task<IActionResult> SubmitScore(string slotType, int id, [FromQuery
                 Type = score.Type,
                 Points = score.Points,
                 SlotId = score.SlotId,
+                AdvSlotId = score.AdvSlotId
             };
 
-            IQueryable<Score> existingScore = this.database.Scores.Where(s => s.SlotId == playerScore.SlotId)
-                .Where(s => s.PlayerIdCollection == playerScore.PlayerIdCollection)
+            IQueryable<Score> existingScore;
+            if(playerScore.AdvSlotId != null) {
+                existingScore = this.database.Scores.Where(s => s.SlotId == playerScore.SlotId && s.AdvSlotId == playerScore.AdvSlotId);
+                
+            }
+            else {
+                existingScore = this.database.Scores.Where(s => s.SlotId == playerScore.SlotId);
+            }
+            existingScore = existingScore.Where(s => s.PlayerIdCollection == playerScore.PlayerIdCollection)
```

Comment:
```suggestion
            IQueryable<Score> existingScore = this.database.Scores.Where(s => s.SlotId == playerScore.SlotId)
                .Where(s => advId != 0 || s.AdvSlotId == advId)
                .Where(s => s.PlayerIdCollection == playerScore.PlayerIdCollection)
                .Where(s => s.Type == playerScore.Type);
```
You can just append another Where statement here instead of splitting it up that will only evaluate the advId comparison if advId is not 0 (not null)